napari/utils/events/_tests/test_evented_model.py

```diff
@@ -16,6 +16,11 @@
 from napari.utils.misc import StringEnum
 
 
+class EventMock(Mock):
+    __enter__ = Mock(return_value=None)
```

Comment:
I might be still waking up, but how does mock `EventEmitter.blocker` as desired?

Reply:
It does not; this is only needed to avoud *errors*, but it does not ensure correct behaviour. This is why the new test actually uses an ad-hoc system.